plugins/inputs/p4runtime/p4runtime_test.go

```diff
@@ -0,0 +1,655 @@
+)
+
+// CounterSpec available here https://github.com/p4lang/p4runtime/blob/main/proto/p4/config/v1/p4info.proto#L289
+func createCounter(
+	name string,
+	id uint32,
+	unit p4ConfigV1.CounterSpec_Unit,
+) *p4ConfigV1.Counter {
+	return &p4ConfigV1.Counter{
+		Preamble: &p4ConfigV1.Preamble{Name: name, Id: id},
+		Spec:     &p4ConfigV1.CounterSpec{Unit: unit},
+	}
+}
+
+func createTelegrafMetric(
+	tags map[string]string,
+	fields map[string]interface{},
+) telegraf.Metric {
+	return testutil.MustMetric("p4_runtime", tags, fields, time.Unix(0, 0))
+}
+
+func createEntityCounterEntry(
+	counterID uint32,
+	index int64,
+	data *p4v1.CounterData,
+) *p4v1.Entity_CounterEntry {
+	return &p4v1.Entity_CounterEntry{
+		CounterEntry: &p4v1.CounterEntry{
+			CounterId: counterID,
+			Index:     &p4v1.Index{Index: index},
+			Data:      data,
+		},
+	}
+}
+
+func NewTestP4RuntimeClient(
+	p4RuntimeClient *fakeP4RuntimeClient,
+	addr string,
+) *P4runtime {
+	conn, _ := grpc.Dial(
+		addr,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	return &P4runtime{
+		Endpoint: addr,
+		DeviceID: uint64(1),
+		Log:      testutil.Logger{},
+		conn:     conn,
+		client:   p4RuntimeClient,
+	}
+}
+
+func TestInitDefault(t *testing.T) {
+	plugin := &P4runtime{Log: testutil.Logger{}}
+	require.NoError(t, plugin.Init())
+	require.Equal(t, "127.0.0.1:9559", plugin.Endpoint)
+	require.Equal(t, uint64(0), plugin.DeviceID)
+	require.Equal(t, 0, len(plugin.CounterNamesInclude))
+	require.Equal(t, false, plugin.EnableTLS)
+}
+
+func TestErrorGetP4Info(t *testing.T) {
+	responses := []struct {
+		getForwardingPipelineConfigResponse      *p4v1.GetForwardingPipelineConfigResponse
+		getForwardingPipelineConfigResponseError error
+	}{
+		{
+			getForwardingPipelineConfigResponse: nil,
+			getForwardingPipelineConfigResponseError: fmt.Errorf(
+				"error when retrieving forwarding pipeline config",
+			),
+		}, {
+			getForwardingPipelineConfigResponse: &p4v1.GetForwardingPipelineConfigResponse{
+				Config: nil,
+			},
+			getForwardingPipelineConfigResponseError: nil,
+		}, {
+			getForwardingPipelineConfigResponse: &p4v1.GetForwardingPipelineConfigResponse{
+				Config: &p4v1.ForwardingPipelineConfig{P4Info: nil},
+			},
+			getForwardingPipelineConfigResponseError: nil,
+		},
+	}
+
+	for _, response := range responses {
+		p4RtClient := &fakeP4RuntimeClient{
+			getForwardingPipelineConfigFn: func(
+				ctx context.Context,
+				in *p4v1.GetForwardingPipelineConfigRequest,
+				opts ...grpc.CallOption,
+			) (*p4v1.GetForwardingPipelineConfigResponse, error) {
+				return response.getForwardingPipelineConfigResponse, response.getForwardingPipelineConfigResponseError
+			},
+		}
+
+		listener, err := net.Listen("tcp", "127.0.0.1:0")
+		require.NoError(t, err)
+
+		plugin := NewTestP4RuntimeClient(p4RtClient, listener.Addr().String())
+
+		var acc testutil.Accumulator
+		errGather := plugin.Gather(&acc)
+		require.Error(t, errGather)
```

Comment:
```suggestion
		require.Error(t, plugin.Gather(&acc))
```
Same for other instances below.

Reply:
done in all occurrences